dlog/plonk-5-wires/src/prover.rs

```diff
@@ -189,12 +189,15 @@ where
             .interpolate()
         });
 
+        let trimmed_length = (index.cs.domain.d1.size as u64).trailing_zeros() as usize;
+        let srs = index.srs.get_ref().trim(trimmed_length);
```

Comment:
here I'm confused as well, why not just use the default SRS? And what if the domain size is not a perfect power of 2?